modules/xmpp/ChatRoom.js

```diff
@@ -1159,8 +1159,13 @@ export default class ChatRoom extends Listenable {
                 this.eventEmitter.emit(XMPPEvents.PRIVATE_MESSAGE_RECEIVED,
                         from, txt, this.myroomjid, stamp);
             } else if (type === 'groupchat') {
+                const nick = $(msg).find('>nick').text();
+
+                // we will fire explicitly that this is a guest(isGuest:true) to the conference
+                // informing that this is probably a message from a guest to the conference (visitor)
+                // a message with explicit name set
                 this.eventEmitter.emit(XMPPEvents.MESSAGE_RECEIVED,
-                        from, txt, this.myroomjid, stamp);
```

Comment:
Doesn't this mark *all* group messages as coming from a visitor?

nit: IMO we should just use "visitor" and avoid introducing "guest"

Reply:
yes 🤦